packages/packages.go

```diff
@@ -28,12 +32,22 @@ type Packages interface {
 // FIXME: Research the various Gopkg name formats and convert them correctly
 func convertGopkgNameToPurl(name string) string {
 	switch {
-	case strings.Contains(name, "github.com"):
-		name = strings.Replace(name, "github.com", "github", 1)
-	case strings.Contains(name, "gopkg.in"):
-		name = strings.Replace(name, "gopkg.in", "github", 1)
-	case strings.Contains(name, "golang.org"):
-		name = strings.Replace(name, "golang.org", "golang", 1)
+	case githubPattern.MatchString(name):
+		// Currently OSS Index's github format support is based on repository
+		// owner/name, so restrict the PURL name path.
+		//
+		// Once golang format support is improved we can switch to that and get
+		// more precise and refined results on repository sub-paths
+		name = githubPattern.ReplaceAllString(name, "github/$1/$2")
+
+	case gopkg2Pattern.MatchString(name):
+		name = gopkg2Pattern.ReplaceAllString(name, "github/$1/$2")
+
+	case gopkg1Pattern.MatchString(name):
+		name = gopkg1Pattern.ReplaceAllString(name, "github/go-$1/$1")
```

Comment:
QUESTION: Not knowing enough here, but this uses the gopkg1Pattern, which is: `"^gopkg.in/([^.]+).*"`, should it be stuffing `github` in the replace?

Reply:
Same as above.

For some additional flavouring, a "proper" golang PURL for github looks like this: `pkg:golang/github.com/<owner>/<name>@<version>`

However, OSS Index has better support for github PURLs than golang PURLs at the moment, though improvements are in progress. For that reason we use github PURLs instead: `pkg:github/<owner>/<name>@<version>`